utils/portworx_configure_max_storage_node_per_zone.sh

```diff
@@ -0,0 +1,165 @@
+    else
+        state=$(kubectl get storagecluster ${PX_CLUSTER_NAME} -n ${NAMESPACE} -o jsonpath='{.status}' | jq)
+        printf "[CHECK PASSED] Portworx Storage Cluster is Online.\n$state\n"
+    fi
+fi
+
+
+
+# Configure kubeconfig
+# Get helm binary over the internet, install helm v3.3.0
+# Trigger the helm upgrade
+
+HELM_VALUES_FILE=/tmp/values.yaml
+
+printf "[INFO] Installing new Helm Charts...\n"
+$CMD repo add ibm-helm https://raw.githubusercontent.com/portworx/ibm-helm/master/repo/stable --force-update
+$CMD repo update
+$CMD get values portworx -n ${CHART_NAMESPACE} > $HELM_VALUES_FILE
+ADVOPT_LINE_NO=$(grep -n 'advOpts' ${HELM_VALUES_FILE} | cut -d ':' -f1)
+if [ "$ADVOPT_LINE_NO" != "" ]; then
+    ADVOPT_LINE=$(grep 'advOpts' ${HELM_VALUES_FILE})
+    REPLACED_LINE=${ADVOPT_LINE//,-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]],/,}
+    REPLACED_LINE=${REPLACED_LINE//-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]],/}
+    REPLACED_LINE=${REPLACED_LINE//,-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]]/}
+    REPLACED_LINE=${REPLACED_LINE//-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]]/}
+
+    REPLACED_LINE=${REPLACED_LINE//;-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]];/;}
+    REPLACED_LINE=${REPLACED_LINE//-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]];/}
+    REPLACED_LINE=${REPLACED_LINE//;-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]]/}
+    REPLACED_LINE=${REPLACED_LINE// ;/ }
+    REPLACED_LINE=${REPLACED_LINE//-max_storage_nodes_per_zone=[[:digit:]|[:digit:][:digit:]|[:digit:][:digit:][:digit:]]/}
+    FIND_AND_REPLACE="${ADVOPT_LINE_NO}s/advOpts.*/${REPLACED_LINE}/"
+    sed -i -e "${FIND_AND_REPLACE}" ${HELM_VALUES_FILE}
+
+    ADVOPT_LENGTH=$(grep "advOpts: " ${HELM_VALUES_FILE} | awk "{print length}")
+    if [ "$ADVOPT_LENGTH" == "9" ]; then
+        FIND_AND_REPLACE="${ADVOPT_LINE_NO}s/.*//"
+        sed -i -e "${FIND_AND_REPLACE}" ${HELM_VALUES_FILE}
+    fi
+else
+    printf "[INFO] advOpts not found in ${HELM_VALUES_FILE}: lno_val:${ADVOPT_LINE_NO}\n"
+fi
+ADVOPT_LINE_NO=$(grep -n 'maxStorageNodesPerZone' ${HELM_VALUES_FILE} | cut -d ':' -f1)
+if [ "$ADVOPT_LINE_NO" != "" ]; then
+    FIND_AND_REPLACE="${ADVOPT_LINE_NO}s/.*//"
+    sed -i -e ${FIND_AND_REPLACE} ${HELM_VALUES_FILE}
+fi
+echo "maxStorageNodesPerZone: ${NODE_COUNT}" >> $HELM_VALUES_FILE
+printf "[INFO] maxStorageNodesPerZone value updated in ${HELM_VALUES_FILE}!!\n"
+printf "[INFO] Upgrading ${HELM_VALUES_FILE}!!\n"
+$CMD upgrade portworx ibm-helm/portworx -f ${HELM_VALUES_FILE} -n ${CHART_NAMESPACE}
+
+if [[ $? -eq 0 ]]; then
+    printf "[INFO] Upgrade Triggered Succesfully, will monitor the storage cluster!!\n"
+else
+    printf "[ERROR] Failed to Upgrade!!\n"
```

Comment:
Upgrade refers to PX version upgrade. Replace `upgrade` keyword everywhere in the file with `configure maxStorageNodesPerZone`.